Runtime/AvatarCreator/Scripts/UI/Elements/AvatarListElement.cs

```diff
@@ -0,0 +1,112 @@
+    {
+        private enum AvatarListFilter
+        {
+            Application,
+            None
+        }
+
+        private const string TAG = nameof(AvatarListElement);
+
+        [SerializeField] private Transform avatarsContainer;
+        [SerializeField] private UserAvatarElement userAvatarElement;
+        [SerializeField] private AvatarListFilter avatarListFilter;
+
+        public UnityEvent<string> onAvatarSelect;
+        public UnityEvent<string> onAvatarModify;
+        public UnityEvent<string> onAvatarDeletionStarted;
+        public UnityEvent<IEnumerable<string>> onAvatarsLoaded;
+
+        private AvatarAPIRequests avatarAPIRequests;
+
+        private readonly Dictionary<string, UserAvatarElement> partnerByAvatarId = new Dictionary<string, UserAvatarElement>();
+
+        public async void LoadAndCreateUserAvatars()
+        {
+            if (!AuthManager.IsSignedIn && !AuthManager.IsSignedInAnonymously)
+            {
+                SDKLogger.Log(TAG, "Not signed in");
+                return;
+            }
+
+            avatarAPIRequests ??= new AvatarAPIRequests();
+
+            var avatarPartnerArr = await avatarAPIRequests.GetUserAvatars(AuthManager.UserSession.Id);
+
+            if (avatarListFilter == AvatarListFilter.Application)
+            {
+                var avatars = avatarPartnerArr.Where((pair) => pair.Value == CoreSettingsHandler.CoreSettings.Subdomain).Select(pair => pair.Key);
+                CreateButtons(avatars);
+            }
+            else
+            {
+                CreateButtons(avatarPartnerArr.Keys);
+            }
+
+        }
+
+        public void RemoveItem(string avatarId)
+        {
+            if (!partnerByAvatarId.ContainsKey(avatarId))
+            {
+                return;
+            }
+
+            Destroy(partnerByAvatarId[avatarId].gameObject);
+            partnerByAvatarId.Remove(avatarId);
+        }
+
+        private void CreateButtons(IEnumerable<string> avatars)
+        {
+            ClearButtons();
+            foreach (var avatarId in avatars)
+            {
+                CreateButton(avatarId);
+            }
+            onAvatarsLoaded?.Invoke(avatars);
+        }
+
+        private void ClearButtons()
+        {
+            if (partnerByAvatarId.Count > 0)
+            {
+                foreach (var avatarListItem in partnerByAvatarId)
+                {
+                    Destroy(avatarListItem.Value.gameObject);
+                }
+            }
+            partnerByAvatarId.Clear();
+        }
+
+        private void CreateButton(string avatarId)
+        {
+            var avatarListElement = Instantiate(userAvatarElement, avatarsContainer).GetComponent<UserAvatarElement>();
+            avatarListElement.SetupButton(avatarId,
+                new UserAvatarElement.AvatarListItemAction { actionToPerform = () => OnAvatarModified(avatarId), actionType = UserAvatarElement.ButtonAction.Customize },
```

Comment:
I still think that doing things this way with, passing 3 separate actions is a bit overcomplicated and harder to read. All 3 buttons have the same event arguments (avatarId) the only difference is that the button action.